core/coreobjects/include/coreobjects/property_object_impl.h

```diff
@@ -723,13 +723,9 @@ ErrCode GenericPropertyObjectImpl<PropObjInterface, Interfaces...>::checkEnumera
         return this->makeErrorInfo(OPENDAQ_ERR_INVALIDSTATE, "Set value is not an enumeration");
 
     EnumerationTypePtr valueEnumerationType = enumerationPtr.getEnumerationType();
-    auto lstEnumerationValues = enumerationPtr.getEnumerationType().getEnumeratorNames();
+    EnumerationTypePtr propEnumerationType = prop.getDefaultValue().asPtrOrNull<IEnumeration>().getEnumerationType();
```

Comment:
A check should be present here for whether or not `prop.getDefaultValue().asPtrOrNull<IEnumeration>()` is assigned. Otherwise this will throw an exception (we should never throw exceptions in interface method implementations, they should return error codes instead).

When confident that the cast is always valid, you can use `.asPtr<>` instead, which will throw an exception if the cast was not successful.

Reply:
Fixed. Thnak you.